include/vapor/FlowParams.h

```diff
@@ -171,35 +246,101 @@ class PARAMS_API FlowParams : public RenderParams {
     //! The rendering type that represents the flow paths.
     //! See RenderType enum class.
     static const std::string RenderTypeTag;
+
     static const std::string RenderRadiusBaseTag;
-    //! Scales the radius of the flow tube rendering
+
+    //! Scales the radius of the flow tube rendering.
+    //! Applies data of type: double.
+    //! Typical values: 0.1 to 5.0.
+    //! Valid values: DBL_MIN to DBL_MAX.
     static const std::string RenderRadiusScalarTag;
-    //! Toggles between rendering 2d glyphs and 3d geometry of the render type
+
+    //! Toggles between rendering 2d glyphs and 3d geometry of the render type.
+    //! Applies data of type: bool.
+    //! Valid values: 0 = off, 1 = on.
     static const std::string RenderGeom3DTag;
+
     static const std::string RenderLightAtCameraTag;
-    //! Draws the direction of the flow stream
+
+    //! Draws the direction of the flow stream.
+    //! Applies data of type: bool.
+    //! Valid values: 0 = off, 1 = on.
     static const std::string RenderShowStreamDirTag;
     
-    //! When rendering samples, determines whether samples are rendered as circles or arrows
+    //! When rendering samples, determines whether samples are rendered as circles or arrows.
+    //! Applies data of type: long.
+    //! Valid values: 0 = FloatParams::GlyphTypeSphere, 1 = FloatParams::GlyphTypeArrow.
     static const std::string RenderGlyphTypeTag;
-    //! When rendering samples, draw every N samples
+
+    //! When rendering samples, draw every N samples.
+    //! Applies data of type: int.
+    //! Typical values: 1 to 20.
+    //! Valid values: INT_MIN to INT_MAX.
     static const std::string RenderGlyphStrideTag;
-    //! When rendering samples, only draw the leading sample in a path
+
+    //! When rendering samples, only draw the leading sample in a path.
+    //! Applies data of type: bool.
+    //! Valid values: 0 = off, 1 = on.
     static const std::string RenderGlyphOnlyLeadingTag;
 
-    //! Falloff parameter for density rendering
+    //! Falloff parameter for density rendering.
```

Comment:
@StasJ can you expand on this explanation? Not sure what "density rendering" means

Reply:
Both of these are somewhat arbitrary parameters to the somewhat arbitrary equation used by the density paper. It's not really a rigorous algorithm, the user is intended to fiddle with them until they get something that looks good. So I don't have much to add unless it's just to try different values and see what looks good. This is easier in the GUI where you have sliders.

Reply:
I've clarified and included a reference to the paper.